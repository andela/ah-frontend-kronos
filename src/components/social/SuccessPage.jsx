import React from 'react';

class SocialLoginSuccess extends React.Component {
  render() {
    return (
      <div>
        <div>
          <h1 className="app">
            {sessionStorage.getItem('user_email')}
            <br />
            Congrats!! You have successfully Logged in with your social Account.
          </h1>
        </div>
      </div>
    );
  }
}

export default SocialLoginSuccess;
