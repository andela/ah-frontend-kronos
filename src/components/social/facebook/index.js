import React, { Component } from 'react';
import FacebookLogin from 'react-facebook-login';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoginFacebook from '../../../actions/socialLogin/FacebookActions';

export class FacebookOption extends Component {
  responseFacebook = (response) => {
    const { LoginFacebook } = this.props;
    LoginFacebook(response.token);
  };

  render() {
    return (
      <div>
        <FacebookLogin
          appId={process.env.FACEBOOK_APP_ID}
          fields="name,email,picture"
          callback={this.responseFacebook}
          icon={<i className="fab fa-facebook-f" />}
          cssClass="my-facebook-button-class"
        />
      </div>
    );
  }
}
FacebookOption.propType = {
  responseFacebook: PropTypes.func,
};
export const mapStateToProps = state => ({
  FacebookAuthState: state.socialLoginReducer,
});

export default connect(
  mapStateToProps,
  { LoginFacebook },
)(FacebookOption);
