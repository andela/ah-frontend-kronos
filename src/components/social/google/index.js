import React, { Component } from 'react';
import GoogleLogin from 'react-google-login';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import LoginGoogle from '../../../actions/socialLogin/GoogleActions';

export class GoogleOption extends Component {
  responseGoogle = (response) => {
    const { LoginGoogle } = this.props;
    const accessToken = LoginGoogle(response.tokenId);
    return accessToken;
  };

  render() {
    return (
      <div className="social-apps">
        <GoogleLogin
          clientId={process.env.GOOGLE_APP_ID}
          buttonText="LOGIN WITH GOOGLE"
          onSuccess={this.responseGoogle}
          onFailure={this.responseGoogle}
          cookiePolicy="single_host_origin"
        />
      </div>
    );
  }
}

GoogleOption.propType = {
  LoginGoogle: PropTypes.func,
};

export const mapStateToProps = state => ({
  googleAuthState: state.socialLoginReducer,
});

export default connect(
  mapStateToProps,
  { LoginGoogle },
)(GoogleOption);
