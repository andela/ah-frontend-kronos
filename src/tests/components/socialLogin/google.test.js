import React from 'react';
import { shallow } from 'enzyme';

import { GoogleOption } from '../../../components/social/google';

describe('Google Login', () => {
  const props = {
    LoginGoogle: jest.fn(),
  };

  const setUp = () => {
    const googleButton = shallow(<GoogleOption {...props} />);
    return googleButton;
  };
  const wrapper = setUp();

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should work as expected', () => {
    const response = {
      tokenId: 'auth_token',
    };
    wrapper.instance().responseGoogle(response);
    expect(props.LoginGoogle).toBeCalled();
  });
});
