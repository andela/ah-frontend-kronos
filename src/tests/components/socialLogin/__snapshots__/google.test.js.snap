// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Login matches the snapshot 1`] = `ShallowWrapper {}`;
